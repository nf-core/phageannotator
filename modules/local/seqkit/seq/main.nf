process SEQKIT_SEQ {
    tag "$meta.id"
    label 'process_low'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/seqkit:2.2.0--h9ee0642_0':
        'biocontainers/seqkit:2.2.0--h9ee0642_0' }"

    input:
    tuple val(meta), path(fastx)

    output:
    tuple val(meta), path("*.fast*")    , emit: fastx
    path "versions.yml"                 , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    def extension = "fastq"
    if ("$fastx" ==~ /.+\.fasta|.+\.fasta.gz|.+\.fa|.+\.fa.gz|.+\.fas|.+\.fas.gz|.+\.fna|.+\.fna.gz/) {
        extension = "fasta"
    }
    def endswith = task.ext.suffix ?: "${extension}.gz"
    """
    seqkit \\
        seq \\
        $fastx \\
        --threads ${task.cpus} \\
        --out-file ${prefix}.${endswith} \\
        $args

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        seqkit: \$( seqkit version | sed 's/seqkit v//' )
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    def extension = "fastq"
    if ("$fastx" ==~ /.+\.fasta|.+\.fasta.gz|.+\.fa|.+\.fa.gz|.+\.fas|.+\.fas.gz|.+\.fna|.+\.fna.gz/) {
        extension = "fasta"
    }
    def endswith = task.ext.suffix ?: "${extension}.gz"
    """
    touch ${prefix}.fasta
    gzip ${prefix}.fasta

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        seqkit: \$( seqkit version | sed 's/seqkit v//' )
    END_VERSIONS
    """
}
