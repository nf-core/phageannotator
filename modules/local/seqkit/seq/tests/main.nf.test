nextflow_process {

    name "Test Process: SEQKIT_SEQ"
    script "../main.nf"
    process "SEQKIT_SEQ"


    test("fasta.gz") {

        when {
            params {
                assembly_min_length = 5
            }
            process {
                """
                input[0] = [
                    [id: 'test'],
                    file(params.pipelines_testdata_base_path + 'modules/local/seqkit/seq/assembly.fasta.gz', checkIfExists: true)
                ]
                """
            }
        }

        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("fasta.gz - stub") {

        options "-stub"

        when {
            params {
                assembly_min_length = 5
            }
            process {
                """
                input[0] = [
                    [id: 'test'],
                    file(params.pipelines_testdata_base_path + 'modules/local/seqkit/seq/assembly.fasta.gz', checkIfExists: true)
                ]
                """
            }
        }

        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}
