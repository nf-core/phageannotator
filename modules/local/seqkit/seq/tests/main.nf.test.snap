{
    "fasta.gz - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.fasta.gz:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,de038a980ca1e484c0e34f4af3f05c2a"
                ],
                "fastx": [
                    [
                        {
                            "id": "test"
                        },
                        "test.fasta.gz:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,de038a980ca1e484c0e34f4af3f05c2a"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-02-19T17:03:57.945355863"
    },
    "fasta.gz": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.fasta.gz:md5,5c3970a6403ad9383b04e0e71ed69ac7"
                    ]
                ],
                "1": [
                    "versions.yml:md5,de038a980ca1e484c0e34f4af3f05c2a"
                ],
                "fastx": [
                    [
                        {
                            "id": "test"
                        },
                        "test.fasta.gz:md5,5c3970a6403ad9383b04e0e71ed69ac7"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,de038a980ca1e484c0e34f4af3f05c2a"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-02-19T17:03:51.240957232"
    }
}
